src/MeshKernelNET/Api/IMeshKernelApi.cs

```diff
@@ -103,6 +103,15 @@ public interface IMeshKernelApi : IDisposable
                                                         in CurvilinearParameters curvilinearParameters,
                                                         in SplinesToCurvilinearParameters splinesToCurvilinearParameters);
 
+        /// <summary>
+        /// Computes the smoothness of a curvilinear grid.
+        /// </summary>
+        /// <param name="meshKernelId">Id of the grid state</param>
+        /// <param name="direction">The direction in which to compute the smoothness (0 for m direction, 1 for n direction)</param>
+        /// <param name="smoothness">The grid smoothness values in the selected direction</param>
+        /// <returns>Error code</returns>
+        int CurvilinearComputeSmoothness(int meshKernelId, int direction, ref double[] smoothness);

```

Comment:
Please add an enum for the direction in CurvilinearDirectionOptions.cs rather than using an int
```
using System.ComponentModel;

namespace MeshKernelNET.Api
{
    public enum CurvilinearDirectionOptions
    {
        [Description("M-direction")]
        M = 0,

        [Description("N-direction")]
        N = 1
    }
}
```